tests/unit_tests/test_ray_tracing.py

```diff
@@ -40,142 +57,102 @@ def ray_tracing_lst(telescope_model_lst, simtel_path, io_handler):
 
     with caplog.at_level(logging.DEBUG):
         ray = RayTracing(
             telescope_model=telescope_model_mst,
             simtel_path=simtel_path,
-            config_data=config_data,
+            zenith_angle=30 * u.deg,
+            source_distance=10 * u.km,
+            off_axis_angle=[0, 2] * u.deg,
+            single_mirror_mode=True,
+            mirror_numbers="all",
         )
 
-    assert ray.config.zenith_angle == 30
+    assert pytest.approx(ray.config.zenith_angle) == 30
     assert len(ray.config.off_axis_angle) == 2
     assert ray.config.single_mirror_mode
     assert "Single mirror mode is activated" in caplog.text
+    assert len(ray.config.mirror_numbers) == telescope_model_mst.mirrors.number_of_mirrors
 
 
 def test_ray_tracing_single_mirror_mode_mirror_numbers(
     simtel_path, io_handler, telescope_model_mst
 ):
-    telescope_model_mst.export_config_file()
-    config_data = {
-        "source_distance": 10 * u.km,
-        "zenith_angle": 30 * u.deg,
-        "off_axis_angle": [0, 2] * u.deg,
-        "single_mirror_mode": True,
-        "mirror_numbers": [1, 2, 3],
-    }
-
     ray = RayTracing(
         telescope_model=telescope_model_mst,
         simtel_path=simtel_path,
-        config_data=config_data,
+        source_distance=10 * u.km,
+        zenith_angle=30 * u.deg,
+        off_axis_angle=[0, 2] * u.deg,
+        single_mirror_mode=True,
+        mirror_numbers=[1, 2, 3],
     )
 
-    assert ray._mirror_numbers == [1, 2, 3]
-
-
-def test_ray_tracing_invalid_telescope_model(simtel_path, io_handler):
-    config_data = {
-        "source_distance": 10 * u.km,
-        "zenith_angle": 30 * u.deg,
-        "off_axis_angle": [0, 2] * u.deg,
-    }
-
-    with pytest.raises(ValueError, match="Invalid TelescopeModel"):
-        RayTracing(
-            telescope_model=None,
-            simtel_path=simtel_path,
-            config_data=config_data,
-        )
+    assert ray.config.mirror_numbers == [1, 2, 3]
 
 
 def test_ray_tracing_read_results(ray_tracing_lst):
     ray_tracing_lst.analyze(force=False)
 
-    assert ray_tracing_lst._has_results is True
     assert len(ray_tracing_lst._results) > 0
     assert ray_tracing_lst.get_mean("d80_cm") == pytest.approx(4.256768651160611, abs=1e-5)
 
 
-def test_export_results(simtel_path, telescope_model_lst, caplog):
+def test_export_results(simtel_path, ray_tracing_lst, caplog):
     """
     Test the export_results method of the RayTracing class without results
     """
 
-    config_data = {
-        "source_distance": 10 * u.km,
-        "zenith_angle": 20 * u.deg,
-        "off_axis_angle": [0, 0] * u.deg,
-    }
-
-    ray = RayTracing(
-        telescope_model=telescope_model_lst,
-        simtel_path=simtel_path,
-        config_data=config_data,
-        label="export_results",
-    )
+    ray = ray_tracing_lst
     ray.export_results()
-    assert "Cannot export results because it does not exist" in caplog.text
+    assert "" in caplog.text
```

Comment:
This doesn't seem useful and I think should be replaced with `No results to export`

Reply:
Yes, fixed.